### PR TITLE
Fixes deleted books actual not being 'deleted'

### LINKED ISSUE #10110: Removed books spawning in random maint.

### PR DESCRIPTION
**Pull request by redmoogle**
# General Documentation

### Intent of your Pull Request

isnull() is not valid SQL believe it or not

fixes: https://github.com/yogstation13/Yogstation/issues/10110

### Why is this change good for the game?

_WGW_

### CODE REVIEW

File: code/modules/library/lib_machines.dm [hunk 1]
```diff
@@ -51,7 +51,7 @@
 				var/datum/DBQuery/query_library_list_books = SSdbcore.NewQuery({"
 				SELECT author, title, category, id
 					FROM [format_table_name("library")]
-					WHERE isnull(deleted)
+					WHERE deleted = NULL
```
Comment (Bibby0110) [inline code/modules/library/lib_machines.dm]:
```suggestion
					WHERE deleted IS NULL
```

File: code/modules/library/lib_machines.dm [hunk 2]
```diff
@@ -100,7 +100,7 @@
 		else
 			author = null
 	if(href_list["search"])
-		SQLquery = "SELECT author, title, category, id FROM [format_table_name("library")] WHERE isnull(deleted) AND "
+		SQLquery = "SELECT author, title, category, id FROM [format_table_name("library")] WHERE deleted = NULL AND "
```
Comment (Bibby0110) [inline code/modules/library/lib_machines.dm]:
```suggestion
		SQLquery = "SELECT author, title, category, id FROM [format_table_name("library")] WHERE deleted IS NULL AND "
```

File: code/modules/library/lib_machines.dm [hunk 3]
```diff
@@ -141,7 +141,7 @@ GLOBAL_LIST(cachedbooks) // List of our cached book datums
 	if(!SSdbcore.Connect())
 		return
 	GLOB.cachedbooks = list()
-	var/datum/DBQuery/query_library_cache = SSdbcore.NewQuery("SELECT id, author, title, category FROM [format_table_name("library")] WHERE isnull(deleted)")
+	var/datum/DBQuery/query_library_cache = SSdbcore.NewQuery("SELECT id, author, title, category FROM [format_table_name("library")] WHERE deleted = NULL")
```
Comment (Bibby0110) [inline code/modules/library/lib_machines.dm]:
```suggestion
	var/datum/DBQuery/query_library_cache = SSdbcore.NewQuery("SELECT id, author, title, category FROM [format_table_name("library")] WHERE deleted IS NULL")
```

File: code/modules/library/lib_machines.dm [hunk 4]
```diff
@@ -463,7 +463,7 @@ GLOBAL_LIST(cachedbooks) // List of our cached book datums
 		else
 			cooldown = world.time + PRINTER_COOLDOWN
 			var/datum/DBQuery/query_library_print = SSdbcore.NewQuery(
-				"SELECT * FROM [format_table_name("library")] WHERE id=:id AND isnull(deleted)",
+				"SELECT * FROM [format_table_name("library")] WHERE id=:id AND deleted = NULL",
```
Comment (Bibby0110) [inline code/modules/library/lib_machines.dm]:
```suggestion
				"SELECT * FROM [format_table_name("library")] WHERE id=:id AND deleted IS NULL",
```

File: code/modules/library/random_books.dm [hunk 5]
```diff
@@ -50,7 +50,7 @@
 	if(prob(25))
 		category = null
 	var/datum/DBQuery/query_get_random_books = SSdbcore.NewQuery(	
-	"SELECT * FROM [format_table_name("library")] WHERE isnull(deleted) AND :category IS NULL OR category = :category ORDER BY rand() LIMIT :limit",
+	"SELECT * FROM [format_table_name("library")] WHERE deleted = NULL AND :category IS NULL OR category = :category ORDER BY rand() LIMIT :limit",
```
Comment (Bibby0110) [inline code/modules/library/random_books.dm]:
```suggestion
	"SELECT * FROM [format_table_name("library")] WHERE deleted IS NULL AND :category IS NULL OR category = :category ORDER BY rand() LIMIT :limit",
```

**Review by Bibby0110 — CHANGES_REQUESTED** [review]
NULL isn't actually a value in SQL, so you can't compare to it. Fun fact, NULL = NULL is FALSE in SQL.

**Review by jcatjmeow — APPROVED** [review]

**Pull request merged**